server/embed/config.go

```diff
@@ -546,6 +547,8 @@ func NewConfig() *Config {
 				Auth:   &clientv3.AuthConfig{},
 			},
 		},
+
+                AutoCompactionMode: DefaultAutoCompactionMode,
```

Comment:
Please also update `Validate` [function](https://github.com/etcd-io/etcd/blob/main/server/embed/config.go#L774).  I think the bug is actually there, because it allows `""` value.
Also, update [parseCompactionRetention](https://github.com/etcd-io/etcd/blob/418010bccb902139ed0d36d5d35fab3bc7d41206/server/embed/etcd.go#L882). It should return error when mode is`""`